### DISCUSSION TITLE
Error while opening file dialog

### DISCUSSION THREAD

**Discussion by Aryan-29** [fltk-rs/fltk-rs#678] [Q&A] [answered]
I am declaring a file dialog with this statement:
let mut webdocs_dialog=FileDialog::new(FileDialogType::BrowseDir);

I have defined a button, by clicking that button webdocs_dialog.show() will be executed.  When I am running my code on my local machine it is running perfectly but after I created a package for my project and ran it on another PC with same dependencies installed I am getting an error mentioned below while clicking the button mentioned above.

Gtk-WARNING **: 10:06:44.549: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Fatal error reading PNG image file: Invalid IHDR data (gdk-pixbuf-error-quark, 0)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Fatal error reading PNG image file: Invalid IHDR data (gdk-pixbuf-error-quark, 0)
Aborted (core dumped)

What might be the issue?

**Comment by MoAlyousef** [comment, marked as answer]
It could be a libpng mismatch between the libpng bundled with fltk and your friend's system, and which is used by gtk over there.
Try building with
```toml
[dependencies]
fltk = { version = "1", features = ["system-libpng"] }
```
You will need libpng-dev(el) on your machine, not on the deployed machine.

**Reply by Aryan-29** [reply]
Thanks that solved the error!
